lib/apps/timers.js

```diff
@@ -50,23 +50,110 @@ class BaseTimer {
 }
 
 class Timer extends BaseTimer {
-    constructor(base, interval) {
+    constructor(base, interval, frequency) {
         super();
 
         this._base = base;
         this._interval = interval;
+        this._frequency = frequency;
     }
 
     toString() {
-        return `[Timer ${this._base}, ${this._interval}]`;
+        return `[Timer ${this._base}, ${this._interval}, ${this._frequency}]`;
     }
 
     _nextTimeout() {
+        // Should we check for cases where frequency > interval?
+        let interval = this._interval;
+        let base = this._base;
         let now = Date.now();
-        if (now < this._base)
-            return this._base - now;
-        let off = (now - this._base) % this._interval;
-        return this._interval - off;
+        let next_timing = null;
+        const DAY = 86400000;
+        const HALFDAY = 43200000;
+        const TIME_9AM = 32400000;
+        const TIME_12PM = HALFDAY;
+
+        if (this._frequency === 0) { // End timer because it will never execute
+            this.stop(); // Not sure if we should use this.end() or this.stop()
```

Comment:
stop is temporary, and means that the timer is paused because the engine is going down (Ctrl-C, kill signal, etc.) or because the rule was paused
end is permanent, and the rule with that timer will self-destruct at the end
here you want end()

Reply:
Ugh sorry how should I fix the merge?